assignment/impl/src/test/org/sakaiproject/assignment/impl/AssignmentComparatorTest.java

```diff
@@ -89,10 +94,49 @@ public void setUp() throws Exception {
 		Mockito.when(userDirectoryService.getUser("user4")).thenReturn(user4);
 		Mockito.when(userDirectoryService.getUser("usernull")).thenReturn(null);
 
+		// Adapted from UserSortNameComparatorTest in kernel
+		final String USERS_WITH_SPACE_PREFIX = "usersWithSpace";
+		User[] usersWithSpace = new User[7];
+		for (int i = 0; i < usersWithSpace.length; i++) {
+			usersWithSpace[i] = Mockito.mock(User.class);
+			when(userDirectoryService.getUser(USERS_WITH_SPACE_PREFIX + i)).thenReturn(usersWithSpace[i]);
+		}
+		when(usersWithSpace[0].getSortName()).thenReturn("Dekfort, Apple");
+		when(usersWithSpace[1].getSortName()).thenReturn("Del Fintino, Pear");
+		when(usersWithSpace[2].getSortName()).thenReturn("Dekford", "Orange");
+		when(usersWithSpace[3].getSortName()).thenReturn("De'Leon", "Cactus");
+		when(usersWithSpace[4].getSortName()).thenReturn("Deleverde", "Mango");
+		when(usersWithSpace[5].getSortName()).thenReturn("Martinez Torcal, Apple");
+		when(usersWithSpace[6].getSortName()).thenReturn("Martin Troncoso, X");
+		for (int i = 0; i < namesWithSpaceSubmissions.length; i++) {
+			namesWithSpaceSubmissions[i] = mockSubmissionForUserID(USERS_WITH_SPACE_PREFIX + i);
+		}
+
 		sortNameComparator = new UserIdComparator(userDirectoryService);
 		submitterNameComparator = new AssignmentSubmissionComparator(assignmentService, siteService, userDirectoryService);
 	}
 
+	/**
+	 * Mock a submission object, as well as its assignment and submitters.
+	 * The assignment will not be a group assignment, and the specified userId will be the only submitter.
+	 */
+	public static AssignmentSubmission mockSubmissionForUserID(String userID)
```

Comment:
No need for a mock here as you can just use `new Assignment`?

Reply:
I wasn't sure, but S/O agrees with you - using real objects seems more future proof. I'll give it a shot

Reply:
Changed to 'createSubmissionForUserID', which now instantiates the intended objects